types/index.d.ts

```diff
@@ -165,37 +208,77 @@ declare namespace unified {
     freeze(): Processor<P>
   }
 
+  /**
+   * A Plugin (Attacher) is the thing passed to `use`.
+   * It configures the processor and in turn can receive options.
+   *
+   * Attachers can configure processors, such as by interacting with parsers and compilers, linking them to other processors, or by specifying how the syntax tree is handled.
+   *
+   * @this Processor context object is set to the invoked on processor.
+   * @param settings Configuration
+   * @param extraSettings Secondary configuration
+   * @typeParam S Plugin settings
+   * @typeParam S2 Extra plugin settings
+   * @typeParam P Processor settings
+   * @returns Optional Transformer.
+   */
   type Plugin<S = Settings, S2 = undefined, P = Settings> = Attacher<S, S2, P>
+
+  /**
+   * Configuration passed to a Plugin or Processor
+   */
   type Settings = {
     [key: string]: unknown
   }
+
   /**
    * Presets provide a potentially sharable way to configure processors.
    * They can contain multiple plugins and optionally settings as well.
+   *
+   * @typeParam P Processor settings
    */
-  interface Preset<S = Settings, P = Settings> {
+  interface Preset<P = Settings> {
     plugins: PluggableList<P>
-    settings?: S
+    settings?: Settings
   }
 
   /**
    * Settings can be passed directly to the processor
+   *
+   * @typeParam P Settings applied to a processor. Useful when packaging unified with a preset parser and compiler.
    */
-  interface ProcessorSettings<S = Settings> {
-    settings: S
+  interface ProcessorSettings<P = Settings> {
+    settings: P
   }
 
-  type PluginTuple<S = Settings, P = Settings> = [Plugin<S, undefined, P>, S]
-  type PluginTriple<S = Settings, S2 = undefined, P = Settings> = [
```

Comment:
`PluginTriple` isn't a great name, merging this into `PluginTuple`, the functionality remains the same.